lib/Memecoin.ts

```diff
@@ -919,21 +966,31 @@ export class MemecoinSDK {
     }
   }
 
-  private async swapCoinFrontend(params: SwapFrontendParams): Promise<HexString> {
-    const { fromToken, toToken, amountIn, amountOut, slippage, affiliate } = params
-
-    const walletClient = this.walletClient
-
-    if (fromToken === 'eth' || toToken === 'eth') {
-      throw new Error('ETH is not supported as a fromToken or toToken')
+  async swap(params: SwapParams): Promise<HexString> {
+    const isBatchSupported = await this.isBatchSupported()
+    if (!isBatchSupported) {
+      await this.switchToBaseChain()
     }
 
-    const { allowance } = params
+    const walletClient = this.walletClient
 
-    const amountOutMin = this.calculateMinAmountWithSlippage(amountOut, slippage)
+    const {
+      amountOutMinimum,
+      tokenIn,
+      amountIn,
+      orderReferrer,
+      recipient,
+      feeIn,
+      feeOut,
+      tokenInPoolType,
+      tokenOutPoolType,
+      tokenOut
+    } = params
+
+    const amountOutMin = this.calculateMinAmountWithSlippage(amountOutMinimum)
```

Comment:
make `calculateMinAmountWithSlippage` a global function in `functions.ts` so you can use it here and in the swapper.ts code